ext/zend_test/tests/strings_marked_as_utf8.phpt

```diff
@@ -0,0 +1,150 @@
+--TEST--
+Check that strings are marked as valid UTF-8
+--EXTENSIONS--
+zend_test
+--FILE--
+<?php
+echo "Empty strings:\n";
+$s = "";
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Known strings:\n";
+$s = "c";
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Integer cast to string:\n";
+$i = 2563;
+$s = (string) $i;
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Float cast to string:\n";
+$f = 26.7;
+$s = (string) $f;
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+$f = 2e100;
+$s = (string) $f;
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Concatenation known valid UTF-8 strings in variables:\n";
+$s1 = "f";
+$s2 = "o";
+$s = $s1 . $s2;
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Multiple concatenation known valid UTF-8 strings in variables:\n";
+$s1 = "f";
+$s2 = "o";
+$s = $s1 . $s2 . $s2;
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Concatenation known valid UTF-8 in assignment:\n";
+$s = "f" . "o";
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+// The "foo" string matches with a "Foo" class which is registered by the zend_test extension.
```

Comment:
interned strings are expected to be heavily used, if the flag cannot be set on the first check (as the string can be shared across threads), the flag should be computed when interning it - thanks to #10436 it should be fast

Reply:
why a down vote?

Reply:
Because this is of no relevance to this PR. Moreover, this computation would be made on every constant string, which will very likely have a measurable performance impact.

Reply:
@alexdowad wdyt? When interned strings are never flagged as valid, they can be effectively the slowest - like class name checked for UTF-8 validity repeatably.

Reply:
> @alexdowad wdyt? When interned strings are never flagged as valid, they can be effectively the slowest - like class name checked for UTF-8 validity repeatably.

Why on earth would you check that a class name is valid UTF-8? But again, like Alex said, this is orthogonal to this PR. This PR is already touching a bunch of things just to get concatenation working (and finding Opcache bugs at the same time...)

Reply:
> Why on earth would you check that a class name is valid UTF-8?

for example in atk4/ui https://github.com/atk4/ui/blob/70361879e024f6801a7795afb9c84dc9055cc367/src/HtmlTemplate/Value.php#L32 the UTF-8 validity is checked for every template value to detect bad input early, rendering some php structure name is not as unusual and can be called hundreds time per page